### DISCUSSION TITLE
Question: Handling of "last action" for Trackables differs between Trackable logging and Trackable inventory logging. Should it stay that way?

### DISCUSSION THREAD

**Discussion by eddiemuc** [cgeo/cgeo#15153] [General]
During refactoring of logging (for log edit/delete) I noticed that we have two different behaviours implemented wrt Trackable action prefill.

("Action" in context of Trackable means the log type e.g. NOTE, VISITED, RETRIEVED, DROPPED and so on)

* If a user opens a trackable and then starts entering a log from there, then the action is prefilled from a global setting. One for all trackables. If I log a trackable A with RETRIEVED then I get RETRIEVED as prefill when I start a log for trackable B
* If a user logs a cache and selects actions for the trackables in his/her inventory, then these actions are prefilled from a setting per-trackable (stored in db_extensions)

Question: is this behaviour intentional? If not, how should it be changed?

Things to consider:
* the action range for inventory logging is very restricted (only DO_NOTHING, VISITED and DROPPED is available) while the action range for trackable logs is much wider (eg. RETRIEVED, GRABBED etc).
* The use cases may be really different:
    * trackable logging is usually used for (yet) unknown trackables e.g. when logging a whole list of trackables from a trip. A global setting is handy there if the whole series should be logged with same action (e.g. VISITED)
    * trackables in inventory however are known and should usually receive the same action on all logged caches. A per-trackable-setting is handy here

One idea would be to combine both. Use following logic in BOTH places:
* If a per-trackable-setting is found, it is used as the prefill
* If not the global setting is used
* If in the current context global setting can't be applied (e.g. inventory logging when global setting is e.g. RETRIEVED) then depending on setting for "autovisit" either DO_NOTHING or VISITED is used

**Comment by Lineflyer** [comment]
> Question: is this behaviour intentional? If not, how should it be changed?

AFAIR it was done intentional.

> If a user opens a trackable and then starts entering a log from there, then the action is prefilled from a global setting. One for all trackables. If I log a trackable A with RETRIEVED then I get RETRIEVED as prefill when I start a log for trackable B

This was done to support users who either grab or discover many trackables in a row.

> If a user logs a cache and selects actions for the trackables in his/her inventory, then these actions are prefilled from a setting per-trackable (stored in db_extensions)

This was implemented based on a feature request to make it possible to handle some TBs in inventory different to others. 
Typical: I want all my inventory to "visit" but one dedicated TB should not "visit" every cache.

The two use cases have a different entry point (dedicated TB log vs. cache log), therefore I think it could stay as it is.
